### ISSUE TITLE
Airflow(2.2.2) UI Does Not Show When DAG Deployment Get Exception And Not Schedules (Related with default_args' owner property)

### ISSUE THREAD

**Issue by bilgehan-biricik** [apache/airflow#21185]
### Apache Airflow version

2.2.2

### What happened

I am new to the Airflow and I am working on a project. I recently made some changes on one of the DAG and added my user name to default_args' owner property like this:
`'owner': ['user.name1', 'user.name2'],`
Me and my teammate merged like this and our CI/CD pipelines finished successfully and I thought deployment was successful. After that in the UI I didn't see my user name and thought changes not reflect immediately. But my DAG's next run column show hours ago and not schedules itself. We start manually once and it run with my latest changes. And I thought my latest changes are passed yet DAG is not schedules itself. Then we check our logs and saw an exception that this owner property is not accept list type but string type. We fix it like this `'owner': 'user.name1,user.name2'` and problem is solved.

TL;DR: When get parse errors on deployment Airflow UI not showing an error and DAG is not schedules itself.

### What you expected to happen

We think UI should show an error message and inform user for this DAG's exception. 

### Operating System

Centos 7

### Deployment

Docker-Compose

### Are you willing to submit PR?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/apache/airflow/blob/main/CODE_OF_CONDUCT.md)

**Comment by potiuk** [comment]
The UI already shows parse errors of DAGs. 

So I am surprised you have not seen it. Probably somoene else saw it and dismissed it - the parsing errors are dismissable and they disappear when clicked.
